app/models/maintenance_tasks/task_data.rb

```diff
@@ -133,17 +133,22 @@ def csv_task?
       !deleted? && Task.named(name).has_csv_content?
     end
 
-    # @return [Array<String>] the names of parameters the Task accepts.
-    def parameter_names
+    # @return [Hash<String, Value>]
+    #   hash of default values indexed by parameter name the Task accepts.
+    def parameters
       if deleted?
-        []
+        {}
       else
-        Task.named(name).attribute_names
+        task_object.attributes
       end
     end
 
     private
 
+    def task_object
+      @task_object ||= Task.named(name).new
```

Comment:
I was considering memoizing `Task.named(name)` lookup as well, in a separate PR, but might be premature

Reply:
If we assign `parameters = @task.parameters`, I don't think memoization is needed.

Reply:
I end up keeping the memoization as `task_object` method became public and we are using it twice (inside `parameter_names` and for `fields_for`)

Reply:
I don't think we need to create a new task instance for `#parameter_names` though since we can just do `Task.named(name).attribute_names` 😄 

I'd prefer not to expose this new method, to leave `#parameter_names` as is, and to only init the Task object in the view:
```erb
<%= form.fields_for :task_arguments, Task.named(@task).new do |ff| %>
```

Reply:
Done!
`task_object = MaintenanceTasks::Task.named(@task.name).new` seems like a little too much code for a view but I feel even more uncomfortable exposing task object and documenting it so let's try initializing it in the view 👍